pandas/tests/indexes/test_multi.py

```diff
@@ -615,8 +615,27 @@ def test_constructor_mismatched_label_levels(self):
         with tm.assert_raises_regex(ValueError, label_error):
             self.index.copy().set_labels([[0, 0, 0, 0], [0, 0]])
 
-    @pytest.mark.parametrize('names', [['a', 'b', 'a'], [1, 1, 2],
-                                       [1, 'a', 1]])
+    def test_constructor_nonhashable_names(self):
+        # GH 20527
+        levels = [[1, 2], [u'one', u'two']]
+        labels = [[0, 0, 1, 1], [0, 1, 0, 1]]
+        names = ((['foo'], ['bar']))
+        message = "MultiIndex.name must be a hashable type"
+        tm.assert_raises_regex(TypeError, message,
+                               MultiIndex, levels=levels,
+                               labels=labels, names=names)
+
+        # With .rename()
+        mi = MultiIndex(levels=[[1, 2], [u'one', u'two']],
+                        labels=[[0, 0, 1, 1], [0, 1, 0, 1]],
+                        names=('foo', 'bar'))
+        renamed = [['foor'], ['barr']]
+        tm.assert_raises_regex(TypeError, message, mi.rename, names=renamed)
+        # With .set_names()
+        tm.assert_raises_regex(TypeError, message, mi.set_names, names=renamed)
+
+    @pytest.mark.parametrize('names', [['a', 'b', 'a'], ['1', '1', '2'],
+                                       ['1', 'a', '1']])
```

Comment:
@arminv Is there a reason that you changed those parametrize values to all strings? (I suppose by accident?) 
I am reworking the test in https://github.com/pandas-dev/pandas/pull/21423, so will revert there if this was by accident

Reply:
@jorisvandenbossche IIRC  I changed it  [(in this commit)](https://github.com/pandas-dev/pandas/pull/20548/commits/b554bb3f050fd9e1aba4cbf02322ddc9fd4d4f41) because the test was failing, but implementation changed a lot after that commit so I'm not sure if reverting this would cause a problem now

Reply:
Seems to be passing there!